codex/contracts/interactions/hostinteractions.nim

```diff
@@ -0,0 +1,67 @@
+import pkg/ethers
+import pkg/chronicles
+
+import ../../sales
+import ../../proving
+import ../../stores
+import ../deployment
+import ../marketplace
+import ../market
+import ../proofs
+import ../clock
+import ./interactions
+
+export sales
+export proving
+export chronicles
+
+type
+  HostInteractions* = ref object of ContractInteractions
+    sales*: Sales
+    proving*: Proving
+
+proc new*(_: type HostInteractions,
+          signer: Signer,
+          deployment: Deployment,
+          repoStore: RepoStore): ?HostInteractions =
+
+  without address =? deployment.address(Marketplace):
+    error "Unable to determine address of the Marketplace smart contract"
+    return none HostInteractions
```

Comment:
Yes, constructors should return `Result` and if they failed to construct the object properly they should error out, returning an `Option` from a constructor is almost always wrong.

Reply:
I definitely agree it's an odd construct. 

Contextually, this PR split `ContractInteractions` into `HostInteractions` and `ClientInteractions`, in an attempt to prevent passing a RepoStore to constructors that don't need it in the constructor. Effectively it is defining two roles for a node: client and host. There could be cli options passed that allow a node to operate only as a host or only as a node, if desired.

I can modify the constructor return types to return a Result as dmitriy suggested.

Reply:
Updated.